src/ui/Omnibox/Omnibox.ts

```diff
@@ -613,51 +620,16 @@ export class Omnibox extends Component {
   }
 
   private buildCustomDataForPartialQueries(index: number, suggestions: string[]): IAnalyticsOmniboxSuggestionMeta {
-    return {
-      partialQueries: this.cleanCustomData(this.partialQueries),
-      suggestionRanking: index,
-      suggestions: this.cleanCustomData(suggestions),
-      partialQuery: _.last(this.partialQueries)
-    };
+    return this.omniboxAnalytics.buildCustomDataForPartialQueries(index, suggestions);
   }
 
-  private cleanCustomData(toClean: string[], rejectLength = 256) {
-    // Filter out only consecutive values that are the identical
-    toClean = _.compact(
-      _.filter(toClean, (partial: string, pos?: number, array?: string[]) => {
-        return pos === 0 || partial !== array[pos - 1];
-      })
-    );
-
-    // Custom dimensions cannot be an array in analytics service: Send a string joined by ; instead.
-    // Need to replace ;
-    toClean = _.map(toClean, partial => {
-      return partial.replace(/;/g, '');
-    });
-
-    // Reduce right to get the last X words that adds to less then rejectLength
-    const reducedToRejectLengthOrLess = [];
-    _.reduceRight(
-      toClean,
-      (memo: number, partial: string) => {
-        const totalSoFar = memo + partial.length;
-        if (totalSoFar <= rejectLength) {
-          reducedToRejectLengthOrLess.push(partial);
-        }
-        return totalSoFar;
-      },
-      0
-    );
-    toClean = reducedToRejectLengthOrLess.reverse();
-    const ret = toClean.join(';');
-
-    // analytics service can store max 256 char in a custom event
-    // if we're over that, call cleanup again with an arbitrary 10 less char accepted
-    if (ret.length >= 256) {
-      return this.cleanCustomData(toClean, rejectLength - 10);
-    }
-
-    return toClean.join(';');
+  private handleQuerySuggestGetFocus(args: IQuerySuggestSelection) {
```

Comment:
Instead of writing args you could use destructuring:
`{ suggestion }: IQuerySuggestSelection`